driver/docker-container/driver.go

```diff
@@ -341,11 +341,18 @@ func (d *Driver) Client(ctx context.Context) (*client.Client, error) {
 		return nil, err
 	}
 
+	clientOpts := []client.ClientOpt{
+		client.WithContextDialer(func(context.Context, string) (net.Conn, error) {
+			return conn, nil
+		}),
+	}
+
 	td, _ := exp.(client.TracerDelegate)
+	if td != nil {
```

Comment:
This should use `if td, ok := ...; ok {` pattern

Reply:
I assume you mean `if td, _ := ...; td {`?

Reply:
No, I think `ok` is correct here and preferred to check for type matches. It would help if you would show where the panic case is happening for you to be sure.

Reply:
Maybe you can share your thought process which went into writing the previous code.

That being said the crash happens in autogenerated code.
So first it extracts the option here: https://github.com/docker/buildx/blob/9a1267cd021083874b02c228374f42da83151c54/vendor/github.com/moby/buildkit/client/client.go#L77

It then passes the reference down and crashes here: https://github.com/docker/buildx/blob/9a1267cd021083874b02c228374f42da83151c54/vendor/github.com/moby/buildkit/client/client.go#L167

since `TracerDelegate` field of `withTracerDelegate` is still set to `nil`.
So IMO constructing `withTracerDelegate` without a valid reference makes the code more opaque (and fails at runtime).

Leaving `ok` out of the current code seems the sensible thing since the path to avoid is the one with having `TracerDelegate` as `nil` and thus checking for `nil` is the most transparent.

Reply:
If in your case the `exp` is nil then just check that directly and skip the option. If for some reason you want to support `WithTracerDelegate(nil)` then it can have a hander inside the function (but I don't think that is actually needed).

Do you have a case where `exp` is not nil and type check will return `nil` with `ok`? I'm not sure how that would be possible.

Reply:
It seems to me like we have a communication problem.
I don't seem to be bringing across my thought process and I don't really understand your response to my posts.
So for now I think I will fork buildx and maintain the patches myself.